### PR TITLE
fix(logging): allow nil zones in resource autodetection

### LINKED ISSUE #3990: logging: panic in resource dection for Cloud Functions
**Client**

Logging

**Environment**

Go 1.13 on Cloud Functions

Updating only the dependency in our project resulted in this panic for our Cloud Function

```
panic: runtime error: slice bounds out of range [:-1]

goroutine 1 [running]:
functions/vendor/cloud.google.com/go/logging.regionFromZone(...)
	/workspace/src/functions/vendor/cloud.google.com/go/logging/logging.go:407
functions/vendor/cloud.google.com/go/logging.detectCloudFunction(0x1379001)
	/workspace/src/functions/vendor/cloud.google.com/go/logging/logging.go:360 +0x2db
functions/vendor/cloud.google.com/go/logging.detectResource.func1()
	/workspace/src/functions/vendor/cloud.google.com/go/logging/logging.go:374 +0x6a
sync.(*Once).doSlow(0x214d888, 0x13c8d80)
	/usr/local/go/src/sync/once.go:66 +0xe3
sync.(*Once).Do(...)
	/usr/local/go/src/sync/once.go:57
functions/vendor/cloud.google.com/go/logging.detectResource(0x120)
	/workspace/src/functions/vendor/cloud.google.com/go/logging/logging.go:367 +0x59
functions/vendor/cloud.google.com/go/logging.(*Client).Logger(0xc00008ef00, 0x1379ef4, 0x7, 0xc000933cb8, 0x2, 0x2, 0xc00027c000)
	/workspace/src/functions/vendor/cloud.google.com/go/logging/logging.go:523 +0x34
```

Related to #3909

### PR DESCRIPTION
**Pull request by nicoleczhu**
fixes #3990

### CODE REVIEW

File: logging/logging.go [hunk 1]
```diff
@@ -404,7 +404,11 @@ func monitoredResource(parent string) *mrpb.MonitoredResource {
 }
 
 func regionFromZone(zone string) string {
-	return zone[:strings.LastIndex(zone, "-")]
+	cutoff := strings.LastIndex(zone, "-")
+	if cutoff > 0 {
+		return zone[:cutoff]
+	}
+	return zone
```
Comment (codyoss) [inline logging/logging.go:411]: If this value is not what is expected should we put zone in at all? Or just `""`?

Comment (someone1) [reply logging/logging.go:411]: Maybe I can help out here, in my experience, passing in the values causing this to panic (e.g. `us16`) will work and be accepted by Cloud Logging. I've been using my own autodetect code to stand up our logging client which does this and have had no issues logging:

image

Comment (nicoleczhu) [reply logging/logging.go:411]: We should accept any user specified/custom zones. Per user docs: 
`CommonResource sets the monitored resource associated with all log entries written from a Logger. If not provided, the resource is automatically detected based on the running environment (on GCE, GCR, GCF and GAE Standard only). This value can be overridden per-entry by setting an Entry's Resource field.`

**Comment by nicoleczhu** [comment]
@codyoss , can I bug you for a stampy stamp on this one?

**Review by codyoss — APPROVED** [review]
Lgtm

**Comment by nicoleczhu** [comment]
> Lgtm

oh wow, thank you for taking a look on your weekend Cody

**Pull request merged**